server/tests/test_workflow_views.py

```diff
@@ -253,3 +254,35 @@ def test_workflow_title_post(self):
         response = workflow_detail(request,  pk=pk_workflow)
         self.assertIs(response.status_code, status.HTTP_200_OK)
         self.assertEqual(response.data['name'], 'Billy Bob Thornton')
+
+
+    def test_workflow_public_post(self):
+        pk_workflow = Workflow.objects.get(name='Workflow 1').id
+        request = self.factory.post('/api/workflows/%d' % pk_workflow,
+                                   {'public': True})
+        force_authenticate(request, user=self.user)
+        response = workflow_detail(request, pk=pk_workflow)
+        self.assertIs(response.status_code, status.HTTP_204_NO_CONTENT)
+
+        # see that we get the new value back
+        request = self.factory.get('/api/wfmodules/%d/' % pk_workflow,)
```

Comment:
I miscommunicated earlier; rather than calling get() this should just check the actual public field
```
workflow =  Workflow.objects.get(name='Workflow 1')
self.assertEqual(workflow.public, False)
pk_workflow = workflow.id
...
workflow.refresh_from_db()
self.assertEqual(workflow.public, True)
```